src/Raven.Client/Documents/Session/AbstractDocumentQuery.cs

```diff
@@ -1546,6 +1546,9 @@ private object TransformValue(WhereParams whereParams, bool forRange = false)
             if (_conventions.TryConvertValueForQuery(whereParams.FieldName, whereParams.Value, forRange, out var strVal))
                 return strVal;
 
+            if (_conventions.TryConvertValueToObjectForQuery(whereParams.FieldName, whereParams.Value, forRange, out var objValue))
```

Comment:
`TryConvertValueToObjectForQuery` vs `TryConvertValueForQuery` - Why do we have both, and how do we explain to users the diff?

Reply:
We can't change `TryConvertValueForQuery`, because of API compact.
But we can use a single location for this and wrap the `TryConvertValueForQuery` callback with one that outputs an object, so we'll only have to check for it once.

Reply:
Yes, this is my concern as well. But I wanted some what more pointers before impl it.